### PR TITLE
login/logout

### PR DESCRIPTION
**Pull request by SakiAkizuki**
- resend password email
- sign in/up
- sign out
- go directly to HomePage when loginState

### CODE REVIEW

**Comment by alivivi** [comment]
great

**Review by alivivi — COMMENTED** [review]
seems like a bit conflict file

File: StudyBuddy/app/src/main/res/layout/activity_reset_password.xml [hunk 1]
```diff
@@ -7,54 +7,38 @@
...
+        app:layout_constraintTop_toBottomOf="@+id/textViewReset"
+        android:hint="Please type your email"
+        tools:layout_editor_absoluteX="8dp" />
 
     <Button
-        android:layout_width="wrap_content"
+        android:layout_width="368dp"
         android:layout_height="wrap_content"
-        android:text="Confirm"
+        android:text="Sent reset Email"
```
Comment (Leksha) [inline StudyBuddy/app/src/main/res/layout/activity_reset_password.xml:35]: Can we just have the button title be "Reset Password" or something like this:
passwordrecovery

**Review by Leksha — COMMENTED** [review]
I guess we can fix the UI later so don't worry about that one :)

**Pull request merged**